src/shim/index.js

```diff
@@ -122,6 +122,18 @@ const commands = {
 function read() {
     let line = system.stdin.readLine();
     if (line) {
+        if (line === 'NOOP') {
+            //if sender has done a noop,
+            //it means sender is not busy
+            //so lets give phantom more time
+            setTimeout(function() {
```

Comment:
Ah, this is the magic bit.  In the comments it's basically mentions it.  If a NOOP request is sent, then the sender is in IDLE mode, in this mode may as well give Phantom a little bit more time.. When the sender is not busy your currently doing setTimeout(read,0), and when the sender gets busy again that's what it will use again.  Also it's only 100ms,  but it's had a massive effect on my idle CPU usage.  Like you said readLine blocks so were been extra nice to the V8 engine during idle.